src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,45 +1,102 @@
 import { NewCommentForm } from '../NewCommentForm';
 import './PostDetails.scss';
 
-export const PostDetails: React.FC = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+type Props = {
+  selectedPostId: number,
+};
+
+export const PostDetails: React.FC<Props> = React.memo(({ selectedPostId }) => {
+  const [post, setPost] = useState<PostItem | null>(null);
+  const [comments, setComments] = useState<CommentInfo[]>([]);
+  const [visibilityOfComments, setvisibilityOfComments] = useState(true);
+
+  const fetchPost = useCallback(async () => {
+    const postItem = await getPost(selectedPostId);
+
+    setPost(postItem);
+  }, []);
+
+  const fetchComments = async () => {
+    if (post) {
+      const commentsFromServer = await getPostComments(post.id);
+
+      setComments(commentsFromServer);
+    }
+  };
+
+  const remove = async (id: number) => {
+    await removeComment(id);
+    await fetchComments();
+  };
+
+  useEffect(() => {
+    fetchPost();
+  }, [selectedPostId]);
+
+  useEffect(() => {
+    fetchComments();
+  }, [selectedPostId]);
+
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+
+      <section className="PostDetails__post">
+        <p>{post?.body}</p>
+      </section>
+      {comments.length > 0
+        ? (
+          <section className="PostDetails__comments">
+            {visibilityOfComments && (
+              <button
+                type="button"
+                className="button"
+                onClick={() => setvisibilityOfComments(!visibilityOfComments)}
+              >
+                {`Hide ${comments.length} comments`}
+              </button>
+            )}
+
+            {!visibilityOfComments && (
+              <button
+                type="button"
+                className="button"
+                onClick={() => setvisibilityOfComments(!visibilityOfComments)}
+              >
+                {`Show ${comments.length} comments`}
+              </button>
+            )}
```

Comment:
You duplicate these buttons, you can shorten this:
```suggestion
          <section className="PostDetails__comments">
              <button
                type="button"
                className="button"
                onClick={() => setvisibilityOfComments(!visibilityOfComments)}
              >
                {visibilityOfComments
	              ? `Show ${comments.length} comments`
	              : `Hide ${comments.length} comments`}
              </button>
```